discovery_child_development/pipeline/openalex_metaflow.py

```diff
@@ -0,0 +1,168 @@
+"""
+works pipeline
+--------------
+
+A pipeline that takes a list of concept IDs and years, and outputs OpenAlex API results.
+
+The thought behind this is to break the results into manageable yearly chunks. For a given year
+and high level concept, the output works may be well over 2GB in size when saved to json.
+"""
+import itertools
+import json
+import requests
+from metaflow import FlowSpec, S3, step, Parameter, retry, batch
+import os
+import boto3
+from dotenv import load_dotenv
+
+# Amend this to your desired concepts/years. OpenAlex allows up to 50 parameters
+# per query, so code is included by default to chunk up the concepts into 40s.
+CONCEPT_IDS = [
+    "C109260823",  # child development
+    "C2993937534",  # childhood development
+    "C2777082460",  # early childhood
+    "C2911196330",  # child rearing
+    "C2993037610",  # child care
+    "C2779415726",  # child protection
+    "C2781192327",  # child behavior checklist
+    "C15471489",  # child psychotherapy
+    "C178229462",  # early childhood education
+    # "C138496976",  # developmental psychology (level 1).
+]
+
+YEARS = [2023]
+
+API_ROOT = "https://api.openalex.org/works?filter="
+
+load_dotenv()
+# Define location to save the file
+S3_BUCKET = os.environ["S3_BUCKET"]
+# subfolder within the bucket
+S3_PATH = "metaflow"
+
+
+def generate_queries(concepts, years):
+    """Generates a list of queries for the list of concepts and
+    years required.
+
+    Args:
+        concepts : list of concepts to be queried
+        years : list of years to be queried
+
+    Returns:
+        query_list : list of all queries
+    """
+    concepts_joined = "|".join(concepts)
+    return [f"{concepts_joined},publication_year:{year}" for year in years]
+
+
+def api_generator(api_root: str, concept_ids: list) -> list:
+    """Generates a list of all URLs needed to completely collect
+    all works relating to the list of concepts.
+
+    Args:
+        api_root : root URL of the OpenAlex API
+        concept_ids : list of concept IDs to be queried
+
+    Returns:
+        all_pages: list of pages required to return all results
+    """
+    concepts_text = concept_ids
+    page_one = f"{api_root}concepts.id:{concepts_text}"
+    print(page_one)
```

Comment:
It would be better to have a more informative output message, eg 
```
f"API request: {page_one)}
```
or something like that...